### PR TITLE
automation part 1

### PR DESCRIPTION
**Pull request by guidotijskens**
first part of automation platform code
provides generation of repo files
provides tool to add rpms to repo files and update repo
not sure if the location is correct / if the keys should be included

### CODE REVIEW

File: automation/configuration/README.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+oci-utils-automation - For Oracle Internal Use Only
+===================================================
+The oci-utils-automation rpms define the repofiles for
+accessing rpms to test and creates the repository
+directories if they do not exist yet.
+
+usage:
+- pull the software tree
+- change directory to the sotware root
+- run 'make publisch'
```
Comment (ejannett) [inline automation/configuration/README.txt:11]: typo publish -> publish

**Comment by ejannett** [comment]
Hi,

I am not sure to understand what's the purpose of these . 
Can you give us more info about the purpose of the change ? Is this about deploying tests , is this deploying infra ? etc...

**Comment by guidotijskens** [comment]
this is the intial part of the setup to automate tests from the oci-automation instance; the oci-automation contains the yum repos (currently OL7 and OL8) for copying our own rpms and eventual others needed to set up tests; this code makes it easy to modify the repo definition if necessary, but more practical to copy rpms-to-test to the repo from where they can be pulled by the test instances...  As oci-automation hosts a webserver, any other data can be provided to tests, but this needs to looked at.
The idea is to spawn a test instance from oci-automation with a configured test suite and let it run on its own and providing a report when done;
the make commands use the username as part of the setup, but maybe this is not enough to enable parallellisation of tests;

**Review by ejannett — APPROVED** [review]

**Comment by guidotijskens** [comment]
obsolete

**Pull request closed without merging**